Add an extrac DST argument to add_{message,service}_files, for actionlib
generated messages, otherwise actionlib message are installed with the full
source path of the build directory under CMAKE_INSTALL_PREFIX.

--- cmake/genmsg-extras.cmake.in~	2012-06-04 23:47:14.000000000 +0200
+++ cmake/genmsg-extras.cmake.in	2012-10-09 18:17:22.081648856 +0200
@@ -26,13 +26,17 @@
 endmacro()
 
 macro(add_message_files)
-  parse_arguments(ARG "PACKAGE;FILES;DIRECTORY" "NOINSTALL;PRIVATE" ${ARGN})
+  parse_arguments(ARG "PACKAGE;FILES;DIRECTORY;DST" "NOINSTALL;PRIVATE" ${ARGN})
   check_unused_arguments("add_message_files" "${ARG_DEFAULT_ARGS}")
 
   if(NOT ARG_PACKAGE)
     set(ARG_PACKAGE ${PROJECT_NAME})
   endif()
 
+  if(NOT ARG_DST)
+    set(ARG_DST ${ARG_DIRECTORY})
+  endif()
+
   string(REGEX MATCH "^/.*" IS_FULL_PATH ${ARG_DIRECTORY})
 
   if(IS_FULL_PATH)
@@ -52,18 +56,21 @@
   list(APPEND ${PROJECT_NAME}_MESSAGE_FILES_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/${ARG_DIRECTORY})
 
   if(NOT ARG_NOINSTALL)
-    install(FILES ${FILES_W_PATH} DESTINATION share/${PROJECT_NAME}/${ARG_DIRECTORY})
+    install(FILES ${FILES_W_PATH} DESTINATION share/${PROJECT_NAME}/${ARG_DST})
   endif()
 endmacro()
 
 macro(add_service_files)
-  parse_arguments(ARG "PACKAGE;FILES;DIRECTORY" "NOINSTALL" ${ARGN})
+  parse_arguments(ARG "PACKAGE;FILES;DIRECTORY;DST" "NOINSTALL" ${ARGN})
   check_unused_arguments("add_service_files" "${ARG_DEFAULT_ARGS}")
 
   if(NOT ARG_PACKAGE)
     set(ARG_PACKAGE ${PROJECT_NAME})
   endif()
 
+  if(NOT ARG_DST)
+    set(ARG_DST ${ARG_DIRECTORY})
+  endif()
 
   string(REGEX MATCH "^/.*" IS_FULL_PATH ${ARG_DIRECTORY})
 
@@ -78,7 +85,7 @@
   list(APPEND ${PROJECT_NAME}_SERVICE_FILES ${FILES_W_PATH})
 
   if(NOT ARG_NOINSTALL)
-    install(FILES ${FILES_W_PATH} DESTINATION share/${PROJECT_NAME}/${ARG_DIRECTORY})
+    install(FILES ${FILES_W_PATH} DESTINATION share/${PROJECT_NAME}/${ARG_DST})
   endif()
 endmacro()
 
